lib/features/mailbox/presentation/mixin/mailbox_widget_mixin.dart

```diff
@@ -248,6 +273,23 @@ mixin MailboxWidgetMixin {
       .toList();
   }
 
+  bool isSubaddressingSupported(Session? session, AccountId? accountId) {
+    if (session == null || accountId == null) {
+      return false;
+    }
+    if (!CapabilityIdentifier.jmapTeamMailboxes.isSupported(session, accountId)) {
+      return false;
+    }
+
+    final subaddressingSupported = ((session
+        .props[0] as Map<CapabilityIdentifier, CapabilityProperties>?)
+        ?[CapabilityIdentifier.jmapTeamMailboxes]
+        ?.props[0] as Map<String, dynamic>?)
+        ?['subaddressingSupported'];
+
+    return subaddressingSupported ?? false;
+  }
+
```

Comment:
Should write test for this. It is not run well with 
```
{
    "capabilities": {
        "urn:apache:james:params:jmap:delegation": {},
        "com:linagora:params:jmap:pgp": {},
        "urn:ietf:params:jmap:websocket": {
            "supportsPush": true,
            "url": "wss://jmap.linagora.com/oidc/jmap/ws"
        },
        "com:linagora:params:jmap:settings": {},
        "com:linagora:params:jmap:ws:ticket": {
            "generationEndpoint": "https://jmap.linagora.com/oidc/jmap/ws/ticket",
            "revocationEndpoint": "https://jmap.linagora.com/oidc/jmap/ws/ticket"
        },
        "urn:apache:james:params:jmap:mail:shares": {},
        "urn:ietf:params:jmap:vacationresponse": {},
        "urn:ietf:params:jmap:mail": {
            "maxMailboxesPerEmail": 10000000,
            "maxMailboxDepth": null,
            "maxSizeMailboxName": 200,
            "maxSizeAttachmentsPerEmail": 20971520,
            "emailQuerySortOptions": [
                "receivedAt",
                "sentAt",
                "size",
                "from",
                "to",
                "subject"
            ],
            "mayCreateTopLevelMailbox": true
        },
        "com:linagora:params:jmap:labels": {},
        "urn:ietf:params:jmap:mdn": {},
        "urn:apache:james:params:jmap:mail:quota": {},
        "urn:ietf:params:jmap:quota": {},
        "urn:apache:james:params:jmap:mail:identity:sortorder": {},
        "urn:ietf:params:jmap:submission": {
            "maxDelayedSend": 0,
            "submissionExtensions": {}
        },
        "com:linagora:params:calendar:event": {
            "replySupportedLanguage": [
                "en",
                "fr"
            ]
        },
        "urn:ietf:params:jmap:core": {
            "maxSizeUpload": 20971520,
            "maxConcurrentUpload": 4,
            "maxSizeRequest": 10000000,
            "maxConcurrentRequests": 4,
            "maxCallsInRequest": 16,
            "maxObjectsInGet": 500,
            "maxObjectsInSet": 500,
            "collationAlgorithms": [
                "i;unicode-casemap"
            ]
        },
        "com:linagora:params:jmap:echo": {},
        "com:linagora:params:jmap:team:mailboxes": {},
        "com:linagora:params:jmap:forward": {},
        "com:linagora:params:jmap:contact:support": {
            "supportMailAddress": null
        },
        "com:linagora:params:jmap:contact:autocomplete": {
            "minInputLength": 2
        },
        "com:linagora:params:jmap:filter": {},
        "com:linagora:params:jmap:firebase:push": {
            "apiKey": null,
            "appId": null,
            "messagingSenderId": null,
            "projectId": null,
            "databaseUrl": null,
            "storageBucket": null,
            "authDomain": null,
            "vapidPublicKey": null
        },
        "com:linagora:params:jmap:public:assets": {
            "publicAssetTotalSize": 20971520
        },
        "com:linagora:params:jmap:messages:vault": {
            "maxEmailRecoveryPerRequest": "5",
            "restorationHorizon": "15 days"
        }
    },
    "accounts": {
        "xxx": {
            "name": "dphamhoang@linagora.com",
            "isPersonal": true,
            "isReadOnly": false,
            "accountCapabilities": {
                "urn:apache:james:params:jmap:delegation": {},
                "com:linagora:params:jmap:pgp": {},
                "urn:ietf:params:jmap:websocket": {
                    "supportsPush": true,
                    "url": "wss://jmap.linagora.com/oidc/jmap/ws"
                },
                "com:linagora:params:jmap:settings": {},
                "com:linagora:params:jmap:ws:ticket": {
                    "generationEndpoint": "https://jmap.linagora.com/oidc/jmap/ws/ticket",
                    "revocationEndpoint": "https://jmap.linagora.com/oidc/jmap/ws/ticket"
                },
                "urn:apache:james:params:jmap:mail:shares": {},
                "urn:ietf:params:jmap:vacationresponse": {},
                "urn:ietf:params:jmap:mail": {
                    "maxMailboxesPerEmail": 10000000,
                    "maxMailboxDepth": null,
                    "maxSizeMailboxName": 200,
                    "maxSizeAttachmentsPerEmail": 20971520,
                    "emailQuerySortOptions": [
                        "receivedAt",
                        "sentAt",
                        "size",
                        "from",
                        "to",
                        "subject"
                    ],
                    "mayCreateTopLevelMailbox": true
                },
                "com:linagora:params:jmap:labels": {},
                "urn:ietf:params:jmap:mdn": {},
                "urn:apache:james:params:jmap:mail:quota": {},
                "urn:ietf:params:jmap:quota": {},
                "urn:apache:james:params:jmap:mail:identity:sortorder": {},
                "urn:ietf:params:jmap:submission": {
                    "maxDelayedSend": 0,
                    "submissionExtensions": {}
                },
                "com:linagora:params:calendar:event": {
                    "replySupportedLanguage": [
                        "en",
                        "fr"
                    ]
                },
                "urn:ietf:params:jmap:core": {
                    "maxSizeUpload": 20971520,
                    "maxConcurrentUpload": 4,
                    "maxSizeRequest": 10000000,
                    "maxConcurrentRequests": 4,
                    "maxCallsInRequest": 16,
                    "maxObjectsInGet": 500,
                    "maxObjectsInSet": 500,
                    "collationAlgorithms": [
                        "i;unicode-casemap"
                    ]
                },
                "com:linagora:params:jmap:echo": {},
                "com:linagora:params:jmap:team:mailboxes": {},
                "com:linagora:params:jmap:forward": {},
                "com:linagora:params:jmap:contact:support": {
                    "supportMailAddress": null
                },
                "com:linagora:params:jmap:contact:autocomplete": {
                    "minInputLength": 2
                },
                "com:linagora:params:jmap:filter": {},
                "com:linagora:params:jmap:firebase:push": {
                    "apiKey": null,
                    "appId": null,
                    "messagingSenderId": null,
                    "projectId": null,
                    "databaseUrl": null,
                    "storageBucket": null,
                    "authDomain": null,
                    "vapidPublicKey": null
                },
                "com:linagora:params:jmap:public:assets": {
                    "publicAssetTotalSize": 20971520
                },
                "com:linagora:params:jmap:messages:vault": {
                    "maxEmailRecoveryPerRequest": "5",
                    "restorationHorizon": "15 days"
                }
            }
        }
    },
    "primaryAccounts": {
        "urn:apache:james:params:jmap:delegation": "xxx",
        "com:linagora:params:jmap:pgp": "xxx",
        "urn:ietf:params:jmap:websocket": "xxx",
        "com:linagora:params:jmap:settings": "xxx",
        "com:linagora:params:jmap:ws:ticket": "xxx",
        "urn:apache:james:params:jmap:mail:shares": "xxx",
        "urn:ietf:params:jmap:vacationresponse": "xxx",
        "urn:ietf:params:jmap:mail": "xxx",
        "com:linagora:params:jmap:labels": "xxx",
        "urn:ietf:params:jmap:mdn": "xxx",
        "urn:apache:james:params:jmap:mail:quota": "xxx",
        "urn:ietf:params:jmap:quota": "xxx",
        "urn:apache:james:params:jmap:mail:identity:sortorder": "xxx",
        "urn:ietf:params:jmap:submission": "xxx",
        "com:linagora:params:calendar:event": "xxx",
        "urn:ietf:params:jmap:core": "xxx",
        "com:linagora:params:jmap:echo": "xxx",
        "com:linagora:params:jmap:team:mailboxes": "xxx",
        "com:linagora:params:jmap:forward": "xxxx",
        "com:linagora:params:jmap:contact:support": "xxx",
        "com:linagora:params:jmap:contact:autocomplete": "xxx",
        "com:linagora:params:jmap:filter": "xxxx",
        "com:linagora:params:jmap:firebase:push": "xxxx",
        "com:linagora:params:jmap:public:assets": "xxx",
        "com:linagora:params:jmap:messages:vault": "xxxx"
    },
    "username": "dphamhoang@linagora.com",
    "apiUrl": "https://jmap.linagora.com/oidc/jmap",
    "downloadUrl": "https://jmap.linagora.com/oidc/download/{accountId}/{blobId}?type={type}&name={name}",
    "uploadUrl": "https://jmap.linagora.com/oidc/upload/{accountId}",
    "eventSourceUrl": "https://jmap.linagora.com/oidc/eventSource?types={types}&closeAfter={closeafter}&ping={ping}",
    "state": "2c9f1b12-b35a-43e6-9af2-0106fb53a943"
}
```

Reply:
please use `session_extension::getCapabilityProperties` to make sure we get properties not only in `capabilities`, but also in `accounts` instance.

Reply:
you `session` attribute does not advertize that subaddressing is supported so in your case `isSubaddressingSupported` should return `false`

would you prefer
```
final subaddressingSupported = (session.getCapabilityProperties(accountId, CapabilityIdentifier.jmapTeamMailboxes) as Map<String, dynamic>)['subaddressingSupported'];
 return subaddressingSupported2 ?? false;
```

Reply:
Sure this session not support subaddressing, but still see the menu action for it. Please check again. Write tests for this too